pkg/apis/camel/v1/trait/deployment.go

```diff
@@ -26,4 +30,19 @@ type DeploymentTrait struct {
 	// The maximum time in seconds for the deployment to make progress before it
 	// is considered to be failed. It defaults to 60s.
 	ProgressDeadlineSeconds *int32 `property:"progress-deadline-seconds" json:"progressDeadlineSeconds,omitempty"`
+	// The deployment strategy to use to replace existing pods with new ones.
+	Strategy appsv1.DeploymentStrategyType `property:"strategy" json:"strategy,omitempty"`
```

Comment:
I wonder if adding `// +kubebuilder:validation:Enum=Recreate;RollingUpdate` would work.

Reply:
added, till I don't recall what to expect :)

Reply:
It worked 😃, the following `enum` has been generated, which makes admission rejecting invalid values:

```yaml
strategy:
  description: The deployment strategy to use to replace existing pods with new ones.
  enum:
    - Recreate
    - RollingUpdate
```